documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -261,4 +266,59 @@ Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rul
+
+**Description**: A tracked resource's `location` property must have the `x-ms-mutability` properties set as `read`, `create`.
+
+**Why the rule is important**: Location is a property that is set once and non-updatable for a tracked resource. Hence, per ARM guidelines the only operations allowed are `read` and `create`.
+
+**How to fix the violation**: Ensure that the `location` property in the tracked resource's hierarchy is correctly set to `read` and `create`.
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="S2008" />S2008	PostOperationIdContainsUrlVerb
+**Output Message**: OperationId must contain the verb: '{0}' in:'{1}'
+
+**Description**: A POST operation's operationID must contain the verb indicated at the end of the corresponding url.
+
+**Why the rule is important**: The url indicates the action performed by it, the corresponding POST operationId must therefore contain this verb for semantic consistency.
+
+**How to fix the violation**: Ensure that the operationId for POST operation contains the verb indicated at the end of the url.
+
+**Good Examples**: Examples of url and corresponding POST operationIds:
+* Url: /foo/{someResource}/activate
+* OperationId: SomeResourceTypes_ActivateResource
+
+**Bad Examples**: Examples of url and corresponding POST operationIds:
+* Url: /foo/{someResource}/activate
+* OperationId: SomeResourceTypes_StartResource
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="S2009" />S2009 ArraySchemaMustHaveItems
+**Output Message**: Please provide an items property for array type: '{0}'.
+
+**Description**: A schema of `array` type must always contain an `items` property. without it, AutoRest will fail to generate an SDK.
+
+**Why the rule is important**: AutoRest needs to know the type of item contained in the array so that it can correctly generate the corresponding code.
+
+**How to fix the violation**: Correctly specify the `items` section for given array type. The items can be of any primitive type or can be a reference to another object type.
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R3011" />R3011 ControlCharactersNotAllowed
+**Output Message**: May not contain control characters:  Characters:'{0}' in:'{1}'
```

Comment:
I don't think I can put the entire list here, I can add a link explaining the control characters instead.